source/event_stream_rpc.c

```diff
@@ -0,0 +1,113 @@
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &stream_id_field)) {
+                *stream_id = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "stream id header value %" PRId32, *stream_id);
+                required_fields_found += 1;
+                goto found;
+            }
+
+            struct aws_byte_buf message_type_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_message_type_name.ptr, aws_event_stream_rpc_message_type_name.len);
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &message_type_field)) {
+                *message_type = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "message type header value %" PRId32, *message_type);
+                required_fields_found += 1;
+                goto found;
+            }
+
+            struct aws_byte_buf message_flags_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_message_flags_name.ptr, aws_event_stream_rpc_message_flags_name.len);
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &message_flags_field)) {
+                *message_flags = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "message flags header value %" PRId32, *message_flags);
+                required_fields_found += 1;
+                goto found;
+            }
+        }
+
+        if (header->header_value_type == AWS_EVENT_STREAM_HEADER_STRING) {
+            struct aws_byte_buf operation_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_operation_name.ptr, aws_event_stream_rpc_operation_name.len);
+
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &operation_field)) {
+                *operation_name = aws_event_stream_header_value_as_string(header);
+                AWS_LOGF_DEBUG(
+                    AWS_LS_EVENT_STREAM_GENERAL,
+                    "operation name header value" PRInSTR,
+                    AWS_BYTE_BUF_PRI(*operation_name));
+                optional_fields_found += 1;
+                goto found;
+            }
+        }
+
+        continue;
+
+    found:
+        if (required_fields_found == 3 && optional_fields_found == 1) {
+            return AWS_OP_SUCCESS;
+        }
+    }
+
+    AWS_LOGF_TRACE(
+        AWS_LS_EVENT_STREAM_GENERAL,
+        "required fields found %zu, optional fields found %zu.",
+        required_fields_found,
+        optional_fields_found);
+    return required_fields_found == 3 ? AWS_OP_SUCCESS : AWS_OP_ERR;
+}
+
+static const uint32_t s_bit_scrambling_magic = 0x45d9f3bU;
+static const uint32_t s_bit_shift_magic = 16U;
+
+/* this is a repurposed hash function based on the technique in splitmix64. The magic number was a result of numerical
+ * analysis on maximum bit entropy. */
+uint64_t aws_event_stream_rpc_hash_streamid(const void *to_hash) {
```

Comment:
Why does this take a `void*` if we know it's an `int32_t*`? Also, if it takes a `void*`, it should probably take a length?

Reply:
this goes into the callback for a hash table, and that's the signature.